repo/content/committed_read_manager.go

```diff
@@ -115,6 +117,81 @@ type SharedManager struct {
 	internalLogger     *zap.SugaredLogger // backing logger for 'sharedBaseLogger'
 }
 
+// loadIndexBlob return index information loaded from the specified blob
+func (sm *SharedManager) loadIndexBlob(ctx context.Context, ibid blob.ID, d gather.WriteBuffer) ([]Info, error) {
+	err := sm.st.GetBlob(ctx, ibid, 0, -1, &d)
+	if err != nil {
+		return nil, err
+	}
+	q, err := ParseIndexBlob(ibid, d.Bytes(), sm.format)
+	if err != nil {
+		return nil, err
+	}
+	return q, nil
+}
+
+// assign store the info struct in a map that can be used to compare indexes
+func assign(iif Info, i int, m map[ID][2]Info) {
+	v := m[iif.GetContentID()]
+	v[i] = iif
+	m[iif.GetContentID()] = v
+}
+
+// ValidateIndexes returns an array of strings (report) that describes differences between
+// the V0 index blob and V1 index blob content.  This is used to check that the upgraded index
+// (V1 index) reflects the content of the old V0 index.
+func (sm *SharedManager) ValidateIndexes(ctx context.Context) ([]string, error) {
+
+	d := gather.WriteBuffer{}
+	indexEntries := map[ID][2]Info{}
+
+	ibis0, _, err := sm.indexBlobManagerV0.listActiveIndexBlobs(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, ibi0 := range ibis0 {
+		// skip the V0 blob poison that is used to prevent client reads.
+		if ibi0.BlobID == legacyIndexPoisonBlobID {
+			continue
+		}
```

Comment:
Should not this check belong in the layer within `listActiveIndexBlobs()` above ? Conceptually, a poison blob is not part of the active index.

cc: @jkowalski

Reply:
@Shrekster agreed.  Note that the method that returns these indices is private, so the intent may be to list _all_ active index blobs of that version (which would necessarily include the poison blob).